lib/core/errors.js

```diff
@@ -147,12 +147,13 @@ class ClientClosedError extends UndiciError {
 }
 
 class SocketError extends UndiciError {
-  constructor (message) {
+  constructor (message, socket) {
     super(message)
     Error.captureStackTrace(this, SocketError)
     this.name = 'SocketError'
     this.message = message || 'Socket error'
     this.code = 'UND_ERR_SOCKET'
+    this.socket = socket
```

Comment:
Where could we document this new property (apart from the TS change?)

Reply:
I can add a new section [here](https://github.com/nodejs/undici/blob/main/docs/api/Errors.md).